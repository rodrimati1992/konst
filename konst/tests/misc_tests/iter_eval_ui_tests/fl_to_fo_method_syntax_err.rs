const R: core::ops::Range<usize> = 0..10;
const S: &[[u8; 2]] = &[[3, 5], [8, 13]];

const _: () = {
    konst::iter::eval!(R, flat_map(), next());
    konst::iter::eval!(R, flat_map(||R), next());
    konst::iter::eval!(R, flat_map(|_|), next());
    konst::iter::eval!(R, flat_map(|_|R, 10), next());
    konst::iter::eval!(R, flat_map(|a, b|R), next());
};

const _: () = {
    konst::iter::eval!(S, flatten(), next());
    konst::iter::eval!(S, flatten(10), next());
};

const _: () = {
    konst::iter::eval!(R, fold());
    konst::iter::eval!(R, fold(0));
    konst::iter::eval!(R, fold(0,));
    konst::iter::eval!(R, fold(0, || 10));
    konst::iter::eval!(R, fold(0, |_| 10));
    konst::iter::eval!(R, fold(0, |_, _| ));
    konst::iter::eval!(R, fold(0, |_, _| 10, 100)); 
};

const _: () = {
    konst::iter::eval!(R, for_each());
    konst::iter::eval!(R, for_each(||None));
    konst::iter::eval!(R, for_each(|_|));
    konst::iter::eval!(R, for_each(|_|(), 100));
    konst::iter::eval!(R, for_each(|a, b|None));
};

fn main(){}
