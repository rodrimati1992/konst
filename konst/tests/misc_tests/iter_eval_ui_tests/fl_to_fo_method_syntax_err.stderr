error: `flat_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:16:5
   |
16 |     konst::iter::eval!(R, flat_map(), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `flat_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:17:5
   |
17 |     konst::iter::eval!(R, flat_map(||R), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `flat_map` expects valid closure syntax, passed closure without return value
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:18:5
   |
18 |     konst::iter::eval!(R, flat_map(|_|), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_no_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `flat_map` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:19:5
   |
19 |     konst::iter::eval!(R, flat_map(|_|R, 10), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `flat_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:20:5
   |
20 |     konst::iter::eval!(R, flat_map(|a, b|R), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `flat_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:21:5
   |
21 |     konst::iter::eval!(R, flat_map(|a, b|R), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `flat_map` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:22:5
   |
22 |     konst::iter::eval!(R, flat_map({whatever1_return::<[u8; 0]>}), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument(s) for `flatten` method: `10`
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:27:5
   |
27 |     konst::iter::eval!(S, flatten(10), next());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument(s) for `fold` method: ``
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:31:5
   |
31 |     konst::iter::eval!(R, fold());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid argument(s) for `fold` method: `0`
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:32:5
   |
32 |     konst::iter::eval!(R, fold(0));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `fold` expects to be passed a 2-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:33:5
   |
33 |     konst::iter::eval!(R, fold(0,));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `fold` expects to be passed a 2-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:34:5
   |
34 |     konst::iter::eval!(R, fold(0, || 10));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `fold` expects to be passed a 2-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:35:5
   |
35 |     konst::iter::eval!(R, fold(0, |_| 10));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `fold` expects valid closure syntax, passed closure without return value
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:36:5
   |
36 |     konst::iter::eval!(R, fold(0, |_, _| ));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_no_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `fold` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:37:5
   |
37 |     konst::iter::eval!(R, fold(0, |_, _| 10, 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `fold` expects to be passed a 2-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:38:5
   |
38 |     konst::iter::eval!(R, fold(0u8, {whatever2_return::<u8>}));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `for_each` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:42:5
   |
42 |     konst::iter::eval!(R, for_each());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `for_each` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:43:5
   |
43 |     konst::iter::eval!(R, for_each(||None));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `for_each` expects valid closure syntax, passed closure without return value
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:44:5
   |
44 |     konst::iter::eval!(R, for_each(|_|));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_no_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `for_each` expects no arguments after closure argument
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:45:5
   |
45 |     konst::iter::eval!(R, for_each(|_|(), 100));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_trailing_expr_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `for_each` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:46:5
   |
46 |     konst::iter::eval!(R, for_each(|a, b|None));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `for_each` expects to be passed a 1-parameter closure
  --> tests/misc_tests/iter_eval_ui_tests/fl_to_fo_method_syntax_err.rs:47:5
   |
47 |     konst::iter::eval!(R, for_each({whatever1_return}));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::__parse_closure_emit_error` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)
